frontend/src/state/ui.ts

```diff
@@ -74,30 +58,35 @@ export class FilterEntry {
 
     @observable isActive = true;
 
-    // Simple
-    @observable property: string = ''; // ex: 'battle.type'
-    @observable operator: FilterOperator = '==';
-    @observable value: string = ''; // any js expression: string, number, or array
-
     // Code
     @observable name: string = ''; // name of the filter, shown instead of the code when set
+    @observable transpiledCode: string = 'return true;\n';
     @observable code: string = 'return true\n//allow all messages'; // js code the user entered
 }
 
-
 export type TimestampDisplayFormat = 'default' | 'unixTimestamp' | 'onlyDate' | 'onlyTime' | 'unixSeconds' | 'relative';
 export function IsLocalTimestampFormat(timestampType: TimestampDisplayFormat) {
     switch (timestampType) {
-        case 'default': return true; // 'localDateTime'
-        case 'onlyDate': return true;
-        case 'onlyTime': return true;
-        case 'relative': return true;
+        case 'default':
+            return true; // 'localDateTime'
+        case 'onlyDate':
+            return true;
+        case 'onlyTime':
+            return true;
+        case 'relative':
+            return true;
     }
     return false;
 }
 
-export enum TopicOffsetOrigin { EndMinusResults = -1, Start = -2, End = -3, Timestamp = -4, Custom = 0 }
-export type TopicMessageSearchSettings = TopicDetailsSettings['searchParams']
+export enum TopicOffsetOrigin {
```

Comment:
Topics don't have offsets, but partitions have. I'd suggest to rename it to `PartitionOffsetOrigin`